frontend/apps/reader/modules/readermenu.lua

```diff
@@ -193,19 +195,19 @@ function ReaderMenu:setUpdateItemTable()
     self.menu_items.exit_menu = {
         text = _("Exit"),
         hold_callback = function()
-            self:exitOrRestart()
+            SetDefaults:saveBeforeExit(function() self:exitOrRestart() end)
```

Comment:
I mean none of this. This is precisely what I meant by "porting the weirdness to the reader" in https://github.com/koreader/koreader/issues/5418#issuecomment-534225444